src/components/RouterLabel/UniswapXRouterLabel.tsx

```diff
@@ -50,12 +50,18 @@ export const UniswapXRouterIcon = () => {
 
 export type UnswapXRouterLabelProps = BoxProps & {
   disableTextGradient?: boolean
+  testId?: string
 }
 
-export default function UniswapXRouterLabel({ children, disableTextGradient, ...rest }: UnswapXRouterLabelProps) {
+export default function UniswapXRouterLabel({
+  children,
+  disableTextGradient,
+  testId,
+  ...rest
+}: UnswapXRouterLabelProps) {
   return (
     <Row gap="xs" width="auto" {...rest} style={{ display: 'inline-flex', ...rest.style }}>
-      <UniswapXRouterIcon />
+      <UniswapXRouterIcon testId={testId} />
```

Comment:
do you need this prop? can you just set the test-id on the svg above without a prop?

Reply:
that would result in all of the uniswapx icons having that testId, making it impossible to check the specific one I'm testing for